### PR TITLE
feat: secondary 4 slices

### PR DESCRIPTION
**Pull request by nehasri89**
Implement Secondary 4 slice for Mobile
https://nidigitalsolutions.jira.com/browse/REPLAT-5140

Remaining bits

- [x] Test
- [x] Use Fixture Generator

### CODE REVIEW

File: packages/edition-slices/edition-slices.showcase.js [hunk 1]
```diff
@@ -22,6 +25,18 @@ export default {
       },
       name: "Lead One And One",
       type: "story"
+    },
+    {
+      component: () => (
+        <SecondaryFour
+          secondary1={secondaryFourData.secondary1}
```
Comment (aronshamash) [inline packages/edition-slices/edition-slices.showcase.js]: as there are 4 tiles of equal hierarchy I think it should be called something else `tile1` etc or `article1` ?

File: packages/edition-slices/edition-tiles.showcase.js [hunk 2]
```diff
@@ -5,18 +5,21 @@ import { PrimaryTile } from "./src/tiles";
 export default {
   children: [
     {
-      component: () => {
-        const slice = mockLeadOneFullWidthSlice();
-        return <PrimaryTile tile={slice.lead} withImage />;
-      },
-      name: "Primary (with image)",
+      component: () => (
+        <PrimaryTile tile={leadOneAndOneData.lead} withImagePosition="top" />
```
Comment (aronshamash) [inline packages/edition-slices/edition-tiles.showcase.js]: think we can drop the `with`

File: packages/edition-slices/src/slices/secondaryfour/index.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { SecondaryFourSlice } from "@times-components/slice-layout";
+import { PrimaryTile } from "../../tiles";
+
+const SecondaryFour = ({ secondary1, secondary2, secondary3, secondary4 }) => (
+  <SecondaryFourSlice
+    renderSecondary1={() => (
```
Comment (aronshamash) [inline packages/edition-slices/src/slices/secondaryfour/index.js]: `renderTile1` etc

File: packages/edition-slices/src/tiles/primary/index.js [hunk 4]
```diff
@@ -46,19 +49,19 @@ const PrimaryTile = ({
...
+    {withImagePosition && withImagePosition === "bottom"
+    ? renderImage(leadAsset.crop169.url)
+    : renderSummaryContent(summary125)}
   </View>
 );
 
 PrimaryTile.propTypes = {
   tile: PropTypes.shape({}).isRequired,
-  withImage: PropTypes.bool
+  withImagePosition: PropTypes.bool
```
Comment (aronshamash) [inline packages/edition-slices/src/tiles/primary/index.js]: looks like this prop is a  `string`

File: packages/slice-layout/src/templates/secondaryfour/index.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import styles from "./styles";
+import propTypes from "./proptypes";
+
+const SecondaryFourSlice = ({
+  renderSecondary1,
+  renderSecondary2,
+  renderSecondary3,
+  renderSecondary4
+}) => {
+  const renderSecondaryRowOne = [renderSecondary1(), renderSecondary2()];
```
Comment (aronshamash) [inline packages/slice-layout/src/templates/secondaryfour/index.js]: don't need `Secondary`

File: packages/edition-slices/edition-slices.showcase.js (hunk 1 above)
Comment (nehasri89) [reply packages/edition-slices/edition-slices.showcase.js]: hmm, atm it is matching with tpa data naming convention

File: packages/edition-slices/src/tiles/primary/index.js [hunk 6]
```diff
@@ -46,19 +49,19 @@ const PrimaryTile = ({
...
 };
 
 PrimaryTile.defaultProps = {
-  withImage: false
+  withImagePosition: false
```
Comment (tuncaulubilge) [inline packages/edition-slices/src/tiles/primary/index.js]: The default should be "null"

File: packages/edition-slices/edition-slices.showcase.js (hunk 1 above)
Comment (aronshamash) [reply packages/edition-slices/edition-slices.showcase.js]: ok stick with TPA naming

File: packages/edition-slices/__tests__/shared-tiles.base.js [hunk 7]
```diff
@@ -2,7 +2,11 @@ import React from "react";
 import TestRenderer from "react-test-renderer";
 import { iterator } from "@times-components/test-utils";
 import leadOneAndOneDataGenerator from "../fixtures/leadoneandone";
-import { PrimaryTile, SecondaryTile } from "../src/tiles";
+import {
+  PrimaryTile,
+  SecondaryTile,
+  SecondaryTileImageReversed
```
Comment (aronshamash) [inline packages/edition-slices/__tests__/shared-tiles.base.js]: think we should sit down and agree tile names @tuncaulubilge @nehasri89  ?

File: packages/edition-slices/edition-slices.showcase.js [hunk 8]
```diff
@@ -2,12 +2,14 @@ import React from "react";
...
-  mockSecondaryOneSlice
+  mockSecondaryOneSlice,
+  mockSecondaryFourSlice
 } from "@times-components/fixture-generator";
 import {
   LeadOneFullWidthSlice,
   LeadOneAndOneSlice,
-  SecondaryOneSlice
+  SecondaryOneSlice,
+  SecondaryFour
```
Comment (aronshamash) [inline packages/edition-slices/edition-slices.showcase.js]: all others have `Slice` at the end so needs that to be consistent although I'm happy to remove from all.

File: packages/fixture-generator/src/mock-slice.ts [hunk 9]
```diff
@@ -56,6 +57,18 @@ function mockSecondaryOneSlice(): SecondaryOneSlice {
   };
 }
 
+function mockSecondaryFourSlice(): SecondaryFourSlice {
+  console.log('inside>>>>>>');
```
Comment (aronshamash) [inline packages/fixture-generator/src/mock-slice.ts]: remove debug

File: packages/edition-slices/__tests__/shared-slices.base.js [hunk 10]
```diff
@@ -1,11 +1,13 @@
...
 import TestRenderer from "react-test-renderer";
 import { iterator } from "@times-components/test-utils";
+import { mockSecondaryFourSlice } from "@times-components/fixture-generator";
 import leadOneAndOneDataGenerator from "../fixtures/leadoneandone";
 import {
   LeadOneFullWidthSlice,
   LeadOneAndOneSlice,
-  SecondaryOneSlice
+  SecondaryOneSlice,
+  SecondaryFour
```
Comment (aronshamash) [inline packages/edition-slices/__tests__/shared-slices.base.js]: still inconsistent slice name

**Pull request merged**